linker/scalajsenv.js

```diff
@@ -48,6 +48,13 @@ const $linkingInfo = {
     "productionMode": false
 //!endif
   },
+//!if moduleKind == NoModule
+  "moduleKind": 0,
+//!else
+//!if moduleKind == CommonJSModule
+  "moduleKind": 1,
+//!endif
+//!endif
```

Comment:
That change, together with the facade in `LinkingInfo.scala`, actually constitutes a user-observable feature. It should be in its own commit, without at the same touching `Platform` and `SystemJSTest`.

Reply:
I'm actually hesitant about exposing this information. I cannot quite figure out what bothers me about it. Maybe the fact that the set of possible values will change with time, unlike the other linking info fields. Or that it might not be meaningful at all for completely different kinds of outputs such as Wasm.

Could we put this information in the `BuildInfo` like fastOpt/fullOpt and `scalaVersion` instead? And then revisit later if we find a user-level use case for knowing `moduleKind` at run-time?

Reply:
Absolutely.